source/backend/dal/Repositories/DispositionFileRepository.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Pims.Dal.Entities;
+using Pims.Dal.Entities.Models;
+using Pims.Dal.Helpers.Extensions;
+
+namespace Pims.Dal.Repositories
+{
+    /// <summary>
+    /// Provides a repository to interact with disposition files within the datasource.
+    /// </summary>
+    public class DispositionFileRepository : BaseRepository<PimsDispositionFile>, IDispositionFileRepository
+    {
+        #region Constructors
+
+        /// <summary>
+        /// Creates a new instance of a DispositionFileRepository, and initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="dbContext"></param>
+        /// <param name="user"></param>
+        /// <param name="logger"></param>
+        public DispositionFileRepository(PimsContext dbContext, ClaimsPrincipal user, ILogger<DispositionFileRepository> logger)
+            : base(dbContext, user, logger)
+        {
+        }
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Retrieves the disposition file with the specified id.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public PimsDispositionFile GetById(long id)
+        {
+            using var scope = Logger.QueryScope();
+
+            return this.Context.PimsDispositionFiles.AsNoTracking()
+                .Include(d => d.DispositionFileStatusTypeCodeNavigation)
+                .Include(d => d.DispositionFundingTypeCodeNavigation)
+                .Include(d => d.DispositionInitiatingDocTypeCodeNavigation)
+                .Include(d => d.DispositionStatusTypeCodeNavigation)
+                .Include(d => d.DispositionTypeCodeNavigation)
+                .Include(d => d.DspInitiatingBranchTypeCodeNavigation)
+                .Include(d => d.RegionCodeNavigation)
+                .Include(d => d.DspPhysFileStatusTypeCodeNavigation)
+                .Include(d => d.PimsDispositionFileTeams)
+                    .ThenInclude(d => d.Organization)
+                .Include(d => d.PimsDispositionFileTeams)
+                    .ThenInclude(d => d.Person)
+                .Include(d => d.PimsDispositionFileTeams)
+                    .ThenInclude(d => d.PrimaryContact)
+                .Include(d => d.PimsDispositionFileTeams)
+                    .ThenInclude(d => d.DspFlTeamProfileTypeCodeNavigation)
+                .FirstOrDefault(d => d.DispositionFileId == id);
```

Comment:
want to confirm this follows existing patterns - I believe we throw a keynotfound exception if no entity is found by id?